facia/app/controllers/front/FrontJsonFapi.scala

```diff
@@ -1,29 +1,41 @@
 package controllers.front
 
 import common.Logging
+import concurrent.{BlockingOperations, FutureSemaphore}
 import conf.Configuration
 import model.PressedPage
-import services.PressedPageService
+import play.api.libs.json.Json
+import services.S3
 import scala.concurrent.{ExecutionContext, Future}
 
 trait FrontJsonFapi extends Logging {
   lazy val stage: String = Configuration.facia.stage.toUpperCase
   val bucketLocation: String
+  val parallelJsonPresses = 24
+  val futureSemaphore = new FutureSemaphore(parallelJsonPresses)
 
-  def pressedPageService: PressedPageService
+  def blockingOperations: BlockingOperations
 
   private def getAddressForPath(path: String, format: String): String = s"$bucketLocation/${path.replaceAll("""\+""", "%2B")}/fapi/pressed.v2.$format"
 
   def get(path: String)(implicit executionContext: ExecutionContext): Future[Option[PressedPage]] = errorLoggingF(s"FrontJsonFapi.get $path") {
-    pressedPageService.get(getAddressForPath(path, "json"))
+    pressedPageFromS3(getAddressForPath(path, "json"))
+  }
+
+  private def pressedPageFromS3(path: String)(implicit executionContext: ExecutionContext): Future[Option[PressedPage]] = errorLoggingF(s"FrontJsonFapi.pressedPageFromS3 $path") {
```

Comment:
Is this private methode necessary? Can its content be moved into `get` above?

Reply:
It can be